atlasdb-config/src/test/java/com/palantir/atlasdb/http/SerializableErrorDecoderTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.http;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.ws.rs.core.MediaType;
+
+import org.apache.http.HttpStatus;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.net.HttpHeaders;
+import com.palantir.lock.LockMode;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.StringLockDescriptor;
+import com.palantir.remoting2.errors.SerializableError;
+import com.palantir.remoting2.ext.jackson.ObjectMappers;
+
+import feign.Response;
+
+@SuppressWarnings("ThrowableNotThrown")
+public class SerializableErrorDecoderTest {
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMappers.newClientObjectMapper();
+
+    private static final String EMPTY_REASON = "";
+    private static final byte[] EMPTY_BODY = new byte[0];
+    private static final String SOME_METHOD_KEY = "METHOD";
+    private static final String EXCEPTION_MESSAGE = "foo";
+    private static final String LOCK_ID = "lock";
+
+    private static final String KNOWN_ERROR_NAME = IllegalArgumentException.class.getCanonicalName();
+    private static final String UNKNOWN_ERROR_NAME = "com.palantir.atlasdb.errors.one.two.three.four";
+    private static final String NOT_ERROR_NAME = "java.lang.Integer";
+
+    private final SerializableErrorDecoder decoder = new SerializableErrorDecoder();
+
+    @Test
+    public void canDecodeExceptionOfKnownType() throws JsonProcessingException {
+        SerializableError serializableError = SerializableError.of(EXCEPTION_MESSAGE, KNOWN_ERROR_NAME);
```

Comment:
The first three tests look exactly the same except for the ERROR_NAME, worth refactoring.